catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchFilterVisitor.java

```diff
@@ -489,6 +490,9 @@ protected static void buildGeometrySearch(BinarySpatialOperator filter, Object d
     } else if (geometryExpression instanceof GeometryImpl) {
       Geometry polygon = ((GeometryImpl) geometryExpression).getJTSGeometry();
       openSearchFilterVisitorObject.addGeometrySearch(polygon);
+    } else if (geometryExpression instanceof MultiPolygon) {
```

Comment:
✏️ This entire method is very procedural, I know you didn't write it. I would suggest pulling our the various geometry-specific logic into separate classes each responsible for unwrapping one geometric type and adding it to the visitor object. That way when more are needed we're not adding yet another `else` block here.

Here's a poor man's sketch:

```
enum GeometryExpressionUpdater {
    SURFACE {
      OpenSearchFilterVisitorObject addGeometry(OpenSearchFilterVisitorObject filterVisitorObject, Literal literal) {
        SurfaceImpl surface = (SurfaceImpl) literal.evaluate(null);
        filterVisitorObject.addGeometrySearch(surface.getJTSGeometry());
        return filterVisitorObject;
      }
    },
    POLYGON {
      OpenSearchFilterVisitorObject addGeometry(OpenSearchFilterVisitorObject filterVisitorObject, Literal literal) {
        filterVisitorObject.addGeometrySearch((Polygon)literal.evaluate(null));
        return filterVisitorObject;
      }
    },
    GEOMETRY {
      OpenSearchFilterVisitorObject addGeometry(OpenSearchFilterVisitorObject filterVisitorObject, Literal literal) {
        Object expression = literal.getValue();
        filterVisitorObject.addGeometrySearch( ((GeometryImpl) expression).getJTSGeometry());
        return filterVisitorObject;
      }
    },
    NOOP {
      OpenSearchFilterVisitorObject addGeometry(OpenSearchFilterVisitorObject filterVisitorObject, Literal literal) {
        return filterVisitorObject;
      }
    };

    public static GeometryExpressionUpdater fromGeometryExpression(Object expression) {
      if (expression instanceof SurfaceImpl) {
        return SURFACE;
      }

      if (expression instanceof Polygon) {
        return POLYGON;
      }

      if (expression instanceof GeometryImpl) {
        return GEOMETRY;
      }

      LOGGER.debug("Unsupported filter constraint");
      return NOOP;
    }

    abstract OpenSearchFilterVisitorObject addGeometry(OpenSearchFilterVisitorObject filterVisitorObject, Literal literal);
  }

```
Then the method becomes something like:

```
Literal literalWrapper = (Literal) filter.getExpression2();
Object geometryExpression = literalWrapper.getValue();
    GeometryExpressionUpdater.fromGeometryExpression(geometryExpression).addGeometry(openSearchFilterVisitorObject, literalWrapper);
```

Reply:
i am not sure the added abstraction is worth it. The new approach still requires code change  for new `geometry` type, and `OpenSearchFilterVisitorObject` continue to be used as input/output parameters. It is novel but introduces additional complexity with more layers to interpret when reading the code.